src/aio/usock_win.inc

```diff
@@ -501,24 +506,20 @@ static void nn_usock_create_io_completion (struct nn_usock *self)
 static void nn_usock_create_pipe (struct nn_usock *self, const char *name)
 {
     char fullname [256];
-
     /*  First, create a fully qualified name for the named pipe. */
     _snprintf(fullname, sizeof (fullname), "\\\\.\\pipe\\%s", name);
 
-    /* TODO: Expose custom nOutBufferSize, nInBufferSize, nDefaultTimeOut,
-             lpSecurityAttributes */
     self->p = CreateNamedPipeA (
         (char*) fullname,
         PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
         PIPE_TYPE_BYTE | PIPE_READMODE_BYTE |
             PIPE_WAIT | PIPE_REJECT_REMOTE_CLIENTS,
         PIPE_UNLIMITED_INSTANCES,
-        4096,
-        4096,
+        self->outbuffersz,
+        self->inbuffersz,
         0,
-        NULL);
+        &self->sec_attr);
 
-    /* TODO: How to properly handle self->p == INVALID_HANDLE_VALUE? */
```

Comment:
Actually, this can fail for a variety of reasons, I think.  Looking at the MSDN site, I see that we should call GetLastError().  We do something like this elsewhere, and indeed a few lines below with CreateFileA we do this.  So we should fix this, or at least leave the comment with TODO in place until we address it.